pkg/scheduler/partition.go

```diff
@@ -967,6 +980,11 @@ func (pc *PartitionContext) GetAppsByState(state string) []*objects.Application
 			appList = append(appList, app)
 		}
 	}
+	for _, app := range pc.completedApplications {
+		if app.CurrentState() == state {
+			appList = append(appList, app)
```

Comment:
We should not range over both lists. If the state is _completed_ we range over the `completedApplications` map in all other cases we range over the `applications` map.